controller/controller.go

```diff
@@ -37,6 +37,10 @@ import (
 const (
 	falseString = "false"
 	trueString  = "true"
+
+	// maxRetries is the number of times a key will be retried before it is dropped out of the queue.
+	// TODO(lichuqiang): make this configurable?
+	maxRetries = 15
```

Comment:
Hmm, this seems strange for a declarative controller.  Are you aware of any other K8s controllers that do this?  rate limiting and back-off, sure, but giving up seems somewhat antithetical.

Reply:
Yeah, some of the k8s controller have this.
Personally I'm fine with both, as the delay would be long enough in the end.

Reply:
removed